core/src/main/java/org/elasticsearch/cluster/metadata/MetaDataIndexAliasesService.java

```diff
@@ -88,12 +88,16 @@ protected ClusterStateUpdateResponse newResponse(boolean acknowledged) {
 
                 @Override
                 public ClusterState execute(ClusterState currentState) {
-                    return innerExecute(currentState, request.actions());
+                    return updateAliasesInClusterState(currentState, request.actions());
                 }
             });
     }
 
-    ClusterState innerExecute(ClusterState currentState, Iterable<AliasAction> actions) {
+    /**
+     * Updates the given cluster state with the requested {@link AliasAction}s to perform
+     * and returns a new cluster state.
+     */
+    public final ClusterState updateAliasesInClusterState(ClusterState currentState, Iterable<AliasAction> actions) {
```

Comment:
I could do that, and the more that I think about it, I think it will be better that way.  It was a bit more complex at the time because I'd have to construct a listener and wait for it to finish, then get the current cluster state from master.  But it may just be the better to do that in this situation, I can encapsulate all my calls to it in one method as a test util.